runtime/runtime-core/common/src/aws/smithy/kotlin/runtime/util/Attributes.kt

```diff
@@ -12,7 +12,10 @@ package aws.smithy.kotlin.runtime.util
  * @param name the name of the attribute (for diagnostics)
  */
 public data class AttributeKey<T>(public val name: String) {
-    override fun toString(): String = if (name.isBlank()) super.toString() else "ExecutionAttributeKey: $name"
+    init {
+        require(name.isNotBlank()) { "AttributeKey name (`$name`) cannot be blank" }
+    }
```

Comment:
**Nit**: If `name` is blank there might not be much value in including it in the error message (e.g., ```"AttributeKey name (``) cannot be blank"```).